test/etc/tasks/system/TouchTaskTest.xml

```diff
@@ -52,11 +54,59 @@
     </target>
 
     <target name="testMappedFilelist">
-        <touch datetime="${DSTAMP}:${TSTAMP}">
+        <touch datetime="${formattedTimestamp}">
             <filelist dir="." files="${tmp.dir}/idonotexist" />
             <mapper type="merge" to="${tmp.dir}/touchtest" />
         </touch>
     </target>
 
+    <target name="testMillis">
+        <touch file="${tmp.dir}/millis-file" millis="946684799864" /> <!-- Friday, December 31, 1999 11:59:59.864 PM -->
```

Comment:
Please test against milliseconds not microseconds.

Reply:
It is milliseconds:
![image](https://user-images.githubusercontent.com/36686538/100695784-d76e7600-3346-11eb-9af4-90fdd94bf6fb.png)